src/core/Track.cpp

```diff
@@ -2200,6 +2392,12 @@ void TrackOperationsWidget::paintEvent( QPaintEvent * pe )
 	{
 		p.drawPixmap( 2, 2, embed::getIconPixmap("track_op_grip_c"));
 	}
+		
+	if( colorBarNeedsUpdate )
+	{
+		colorBarNeedsUpdate = false;
+		update();
+	}
```

Comment:
It's not clear to me what this is for. The colour bar is already painted earlier in `paintEvent`; this will just cause the `TrackOperationsWidget` to be painted a second time.

Reply:
There is a bug where the `TrackOPWidget` has to be painted twice, as it doesn't update with the color changes the first time it's painted. This results in the track displaying a different color than it actually is, and causing an update (eg. moving the track) changes it to the real color. I don't know what causes this, but repainting it fixes the issue, so I stuck with it.

Reply:
I tested this locally and it seems sufficient to call `update()` instead in the places where you set `colorBarNeedsUpdate` to true. Then the `colorBarNeedsUpdate` variable can go.

When the menu closes, it triggers a repaint of the `TrackOperationsWidget`, but only the region where the cog button is. You're asking for a full repaint when you get this partial repaint event, but you can get the same effect by asking for a full repaint in the first place.

Reply:
Yep, can confirm that is the case.

I'll be committing soon to remove this variable.